pkg/scheduler/plugins/proportion/proportion.go

```diff
@@ -178,73 +177,36 @@ func (pp *proportionPlugin) OnSessionOpen(ssn *framework.Session) {
-				attr.name, attr.weight, totalWeight)
-			if _, found := meet[attr.queueID]; found {
-				continue
-			}
-
-			oldDeserved := attr.deserved.Clone()
-			attr.deserved.Add(remaining.Clone().Multi(float64(attr.weight) / float64(totalWeight)))
-
-			if attr.realCapability != nil {
-				attr.deserved.MinDimensionResource(attr.realCapability, api.Infinity)
-			}
-			attr.deserved.MinDimensionResource(attr.request, api.Zero)
-
-			klog.V(4).Infof("Format queue <%s> deserved resource to <%v>", attr.name, attr.deserved)
-
-			if attr.request.LessEqual(attr.deserved, api.Zero) {
-				meet[attr.queueID] = struct{}{}
-				klog.V(4).Infof("queue <%s> is meet", attr.name)
-			} else if reflect.DeepEqual(attr.deserved, oldDeserved) {
-				meet[attr.queueID] = struct{}{}
-				klog.V(4).Infof("queue <%s> is meet cause of the capability", attr.name)
-			}
-			attr.deserved = helpers.Max(attr.deserved, attr.guarantee)
-			pp.updateShare(attr)
-
-			klog.V(4).Infof("The attributes of queue <%s> in proportion: deserved <%v>, realCapability <%v>, allocate <%v>, request <%v>, elastic <%v>, share <%0.2f>",
-				attr.name, attr.deserved, attr.realCapability, attr.allocated, attr.request, attr.elastic, attr.share)
-
-			increased, decreased := attr.deserved.Diff(oldDeserved, api.Zero)
-			increasedDeserved.Add(increased)
-			decreasedDeserved.Add(decreased)
-
-			// Record metrics
-			metrics.UpdateQueueDeserved(attr.name, attr.deserved.MilliCPU, attr.deserved.Memory)
-		}
+	for _, attr := range pp.queueOpts {
+		totalShareResource.Add(attr.guarantee)
+		subResource := helpers.Max(attr.allocated, attr.guarantee)
+		subResource = helpers.Min(subResource, totalShareResource)
+		totalShareResource.Sub(subResource)
+	}
 
-		remaining.Sub(increasedDeserved).Add(decreasedDeserved)
-		klog.V(4).Infof("Remaining resource is  <%s>", remaining)
-		if remaining.IsEmpty() || reflect.DeepEqual(remaining, oldRemaining) {
-			klog.V(4).Infof("Exiting when remaining is empty or no queue has more reosurce request:  <%v>", remaining)
-			break
-		}
+	for _, attr := range pp.queueOpts {
+		realGuarantee := helpers.Max(attr.allocated, attr.guarantee)
+		// the resource at least guarantee
+		deserved := helpers.Max(attr.request, realGuarantee)
+		// the resource can not more than real capability
+		deserved = helpers.Min(deserved, attr.realCapability)
+		// the resource exceed guarantee will occupy total share resource
+		extraResource := deserved.Clone().Sub(realGuarantee)
+		// extra resource can not exceed total share resource
+		extraResource = helpers.Min(extraResource, totalShareResource)
+		// finally resource = guarantee + extraResource
+		attr.deserved = realGuarantee.Clone().Add(extraResource)
+		// total share resource sub extraResource
+		totalShareResource.Sub(extraResource)
+		//attr.deserved = helpers.Max(attr.deserved, attr.guarantee)
+		pp.updateShare(attr)
+
+		klog.V(3).Infof("The attributes of queue <%s> in proportion: deserved <%v>, realCapability <%v>, "+
```

Comment:
Let's keep the log lever lower than `v3` to enhance the performance.

Reply:
I sumbit a new file to replace previous code, the path: pkg/scheduler/plugins/guaranteequota/guaranteequota.go